core/src/test/java/io/questdb/griffin/SortAndLimitTest.java

```diff
@@ -0,0 +1,195 @@
+                "insert into sorttest select x from long_sequence( 10000 );");
+
+        assertQuery("i\n1\n2\n3\n4\n5\n", "select i from sorttest order by i limit 5");
+    }
+
+    @Test
+    public void testInsert_10k_records_AndSelect_Top_5_returns_last_5_records() throws Exception {
+        runQueries(
+                "create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 10000 );");
+
+        assertQuery("i\n9996\n9997\n9998\n9999\n10000\n", "select i from sorttest order by i limit -5");
+    }
+
+    @Test
+    public void testInsert_1k_records_And_Select_Middle_5() throws Exception {
+        runQueries(
+                "create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 1000 );");
+
+        assertQuery("i\n995\n996\n997\n998\n999\n", "select i from sorttest order by i limit 994, -1");
+    }
+
+    @Test
+    public void testInsertAndSelect_Bottom_5_returns_0_records_because_output_is_empty() throws Exception {
+        runQueries("create table sorttest(i int);");
+
+        assertQuery("i\n", "select i from sorttest order by i limit 3");
+    }
+
+    @Test
+    public void testInsertAndSelect_Bottom_5_returns_first_3_records_because_output_is_smaller_than_limit() throws Exception {
+        runQueries(
+                "create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 3 );");
+
+        assertQuery("i\n1\n2\n3\n", "select i from sorttest order by i limit 3");
+    }
+
+    @Test
+    public void testInsertAndSelect_Bottom_5_returns_first_5_records() throws Exception {
+        runQueries("create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 10 );");
+
+        assertQuery("i\n1\n2\n3\n4\n5\n", "select i from sorttest order by i limit 5");
+    }
+
+    @Test
+    public void testInsert_with_duplicates_AndSelect_Bottom_5_returns_first_5_records() throws Exception {
+        runQueries(
+                "create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 10 );",
+                "insert into sorttest select x from long_sequence( 10 );");
+
+        assertQuery("i\n1\n1\n2\n2\n3\n", "select i from sorttest order by i limit 5");
+    }
+
+    @Test
+    public void testInsert_with_duplicates_AndSelect_Top_5_returns_last_5_records() throws Exception {
+        runQueries("create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 10 );",
+                "insert into sorttest select x from long_sequence( 10 );");
+
+        assertQuery("i\n8\n9\n9\n10\n10\n", "select i from sorttest order by i limit -5");
+    }
+
+    @Test
+    public void testInsertAndSelect_Top_5_returns_last_5_records() throws Exception {
+        runQueries("create table sorttest(i int);",
+                "insert into sorttest select x from long_sequence( 10 );");
```

Comment:
Would be good if you re-shuffle the records to test that sort works.
Add designated timestamp column and put random values within same partition before the insert, it will randomise the order.

Reply:
I added these kind of tests for both DESC and ASC sort order to SortAndLimiTest .